Fix Belarusian Cyrillic layout & add Łacinka

lomtjjz opened the pull request:
Hi. This PR fixes some errors with Belarusian Cyrillic layout and introduces an option for a Latin-based script, ([Łacinka](https://en.wikipedia.org/wiki/Belarusian_Latin_alphabet)).


### Changes in [`Cyrillic/belarusian.yaml`](Cyrillic/belarusian.yaml):
- `ў`: Added to the layout.
- `щ`: Removed, as it does not exist in the Belarusian alphabet. Both Xorg and AOSP map `ў` to this key.
- `є`, `и`, and `ї`: Removed, as these letters are specific to the Ukrainian alphabet and are not part of Belarusian.

Also I renamed the file to [`Cyrilic/belarusian_cyrillic.yaml`](Cyrilic/belarusian_cyrillic.yaml).


### Łacinka
- See [`LatinScript/belarusian_latin.yaml`](LatinScript/belarusian_latin.yaml)

tenextractor commented:
Sorry, I forgot to include Ў in the layout. But why remove И Ї Є? They are not top level keys, only long press options, and don't cause any problems. Even in the English keyboard you can get letters like Á even though they are not part of the English alphabet.

lomtjjz commented:
"Less is better?"

<br>

But yes, you're right. Cross-compatibility across layouts is good... and following this thought, we also need to consider adding  `щ` and `ъ`

tenextractor commented:
Ъ is automatically added in Cyrillic layouts under Ь, so we just need Щ under Ш

lomtjjz commented:
Oh, ok. However, I would argue that placing `Щ` under `Ў` could be just as intuitive. Maybe the best solution is to map it under both `Ш` and `Ў`

tenextractor commented:
Yes, it could be under both

tenextractor commented:
And about the Latin layout, making it long press based is bad for typing speed, it could be something like this:
```
ć č ź ž ń ŭ ś š ł
q w e r t y u i o p
a s d f g h j k l
z x c v b n m
```
This might be too big, but we don't have a proper layer system which could shrink the layout, so this is the best possible layout at the moment.

lomtjjz commented:
I never saw such a thing done before for the Łacinka. It will definitely be confusing for the first timers

tenextractor commented:
Yes this is fine, and we can simplify the layout YAML:
```
name: Belarusian (Latin)
languages: be-Latn
rows:
  - letters: ź ž ś š ć č ŭ ĺ ł ń
  - letters: 
    - q
    - w
    - e
    - r
    - t
    - y
    - ['u', 'ŭ']
    - i
    - o
    - p
  - letters:
    - a
    - ['s', 'ś', 'š']
    - d
    - f
    - g
    - h
    - j
    - k
    - ['l', 'ĺ', 'ł'] 
  - letters:
    - ['z', 'ź', 'ž']
    - x
    - ['c', 'ć', 'č']
    - v
    - b
    - ['n', 'ń']
    - m
```

lomtjjz commented:
Hi. Are there any further changes needed or is it ready to be merged?

abb128 commented on mapping.yaml:
```diff
@@ -33,7 +33,9 @@ languages:
   be_BY:
-    - belarusian
+    - belarusian_cyrillic
+    - belarusian_latin
+    - belarusian_latin2
```
The latin layouts should be for a separate language, `be_Latn`

lomtjjz replied:
Done

lomtjjz commented:
Hi. Can you review it once again? @abb128

lomtjjz commented:
Hello again. I [fixed](https://github.com/futo-org/futo-keyboard-layouts/commit/3932265d6058dcb9b20b40c1774988067d75a4fc) the problem mentioned in the latest [review](https://github.com/futo-org/futo-keyboard-layouts/pull/124#pullrequestreview-2479698292). Is there anything else that needs to be addressed before merging this pr?

lomtjjz commented:
https://youtu.be/5NnxJ7K8sDQ

abb128 commented on Cyrillic/belarusian_cyrillic.yaml:
Don't rename the layout file. Internally the enabled layouts are stored by filename, so changing `belarusian` to `belarusian_cyrillic` means the belarusian layout will break for existing users

lomtjjz commented:
Hi. Can you make a review?

lomtjjz commented:
@abb128

lomtjjz commented:
Hi. Here is a brief overview of the changes introduced in this PR:


* [`Cyrillic/belarusian.yaml`](<https://github.com/futo-org/futo-keyboard-layouts/pull/124/files#diff-bda9de3b7aee6e00e0e8401dde48f4530e70c52701bf47e1cfc31cc7ace06d1c>): Added the letter `ў` (U short) in place of `щ` (which is not used in Belarusian btw). The `щ` key can still be accessed via a long press for compatibility with other layouts
* [`LatinScript/belarusian_latin.yaml`](<https://github.com/futo-org/futo-keyboard-layouts/pull/124/files#diff-fb274b5b89e71c917f511dce42d1868a204847b15807d8368af70ef51bfc32cf>) and [`LatinScript/belarusian_latin2.yaml`](<https://github.com/futo-org/futo-keyboard-layouts/pull/124/files#diff-5caeed0d122a93410ed90f8e3c2a52093d0bc1077dedf202ae316993c13719bc>): Introduced two new layouts for Łacinka. The `belarusian_latin2` layout includes an additional row with special letters for faster typing, similar to the Czech layout
* [`mapping.yaml`](<https://github.com/futo-org/futo-keyboard-layouts/pull/124/files#diff-ea93d45aab50f3b6f035d0310960dc7607f664ded1a78aa765b49fad0087e6a7>): Introduced a new category, `be_Latn`

Let me know if there is anything I can do to make it easier for you. 

Before merging, I would appreciate discussing a few points. Is there a preferred way to contact you? Email, Discord, Zulip, or perhaps something else?

Awaiting your reply

abb128 commented:
Thank you for the PR and for your patience

lomtjjz commented:
Thanks for merging this. I had some questions that were cleared up with the app update. Have a good day
